sklearn/datasets/svmlight_format.py

```diff
@@ -179,7 +179,6 @@ def _open_and_load(f, dtype, multilabel, zero_based, query_id,
         actual_dtype, data, ind, indptr, labels, query = \
             _load_svmlight_file(f, dtype, multilabel, zero_based, query_id,
                                 offset, length)
-    # XXX remove closing when Python 2.7+/3.1+ required
```

Comment:
Even if it works it CPython, relying on the garbage collection to close file handlers is bad: it won't work with PyPy.